test/functional/page_objects/common_page.js

```diff
@@ -405,6 +405,24 @@ export function CommonPageProvider({ getService, getPageObjects }) {
         return await jsonElement.getVisibleText();
       }
     }
+
+    /**
+     * Helper to detect an OSS licensed Kibana
+     * Useful for functional testing in cloud environment
+     */
+    async isOss() {
+      const baseUrl = this.getEsHostPort();
+      const username = config.get('servers.elasticsearch.username');
+      const password = config.get('servers.elasticsearch.password');
+      const response = await fetch(baseUrl + '/_xpack', {
+        method: 'get',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': 'Basic ' + Buffer.from(username + ':' + password).toString('base64')
+        },
+      });
+      return response.status !== 200;
+    }
```

Comment:
@LeeDr this new method might be interesting for you, remember you talking about cloud-test-detection. for much issues the detection whether it's an OSS kibana might be helpful

Reply:
Thanks!  Yes, there is a case in another test where I added a check for the host name to see if it's `localhost` or not.  I assumed if it's not localhost then it's running on Cloud.  But now I realize there's really 3 cases for some functionality.
1. OSS
2. Default Distribution (which has x-pack features but not on Cloud)
3. Cloud.  One difference between #2 and #3 is the anonymous user on Cloud.  There may be others.